amd/src/courseformat/tabtree.js

```diff
@@ -0,0 +1,208 @@
+        this.selectors = {
+            TAB: `ul:first-of-type li`,
+            CHILDTAB: `ul:nth-child(2) li`
+        };
+        // Default classes
+        this.classes = {
+            ACTIVETAB: 'active'
+        };
+        // Objects to keep tabs on the tabs
+        this.tabs = {};
+        this.childtabs = {};
+        this.activetab = [null, null];
+    }
+
+    getWatchers() {
+        return [
+            // Sections sorting.
+            {watch: `course.sectionlist:updated`, handler: this._refreshCourseSectionTabs},
+        ];
+    }
+
+    static init(target) {
+        return new Component({
+            element: document.getElementById(target),
+            reactive: getCurrentCourseEditor()
+        });
+    }
+
+    /**
+     * Initial state ready method.
+     *
+     */
+    stateReady() {
+        // Get tab elements.
+        const tabs = this.getElements(this.selectors.TAB);
+        for (let i = 0; i < tabs.length - 1; i++) { // Don't count last "add section" tab.
+            let tab = tabs.item(i);
+            let id = tab.querySelector("a div").dataset.itemid;
+            let classes = tab.querySelector("a").classList.value;
+            this.tabs[id] = tab;
+            if (classes.indexOf(this.classes.ACTIVETAB) !== -1) {
+                this.activetab[0] = id;
+                this.activetab[1] = id;
+            }
+        }
+
+        const childtabs = this.getElements(this.selectors.CHILDTAB);
+        for (let i = 0; i < childtabs.length - 1; i++) { // Don't count last "add section" tab.
+            let tab = childtabs.item(i);
+            let id = tab.querySelector("a div").dataset.itemid;
+            let classes = tab.querySelector("a").classList.value;
+            this.childtabs[id] = tab;
+            if (classes.indexOf(this.classes.ACTIVETAB) !== -1) {
+                this.activetab[1] = id;
+            }
+        }
+    }
+
+    /**
+     * Refresh the section tabs.
+     *
+     * @param {object} param
+     * @param {Object} param.element
+     */
+    _refreshCourseSectionTabs({element}) {
+        // Change the active top-level tab, if necessary.
+        const activeTab1 = this.reactive.get('section', this.activetab[1]);
+        let newActiveTab0id = (activeTab1.levelsan >= 1) ? activeTab1.parentid : activeTab1.id;
+        if (newActiveTab0id != this.activetab[0]) {
+            let section = this.reactive.stateManager.state.section.get(this.activetab[0]);
+            let anchor = this.element.querySelector('ul:first-of-type div[data-itemid="' + this.activetab[0] + '"]').parentElement;
+            anchor.classList.remove("active");
+            anchor.href = section.sectionurl.replace("&amp;", "&");
+            this.activetab[0] = newActiveTab0id;
+            anchor = this.element.querySelector('ul:first-of-type div[data-itemid="' + this.activetab[0] + '"]').parentElement;
+            anchor.classList.add("active");
+            anchor.removeAttribute("href");
+        }
```

Comment:
Sorry, I forgot to change the add section tab to match.
```suggestion
            const addAnchor = this.element.querySelector('ul:nth-of-type(2) li:last-of-type a');
            const addLink = addAnchor.href.replace(/&insertparentid=\d+/, "&insertparentid=" + this.activetab[0]);
            addAnchor.setAttribute("href", addLink);
        }
```